contents/ruby/Rails.md

```diff
@@ -0,0 +1,169 @@
+<a href="">URI</a>
+</tooltip> in the `routes.rb` file, which defines, by default, all the routes in the application. Each valid URI is mapped to a function located inside a Controller, that is then invoked to provide a response to the request. 
+
+In order to achieve this, Rails exposes a convenient abstraction called a "Resource". A resource can refer to an object upon which you can perform CRUD (create, read, update, and delete) operations. These operations are often invoked via a HTTP request of the appropriate request verb (POST, GET, PUT, and DELETE respectively). In order to declare a resource called `user`, you would mention it inside your `routes.rb` in the following manner: 
+
+```ruby
+resources :users
+```
+
+Now, you can check all the routes that you have in your application by running `rails routes` in the command line. This should include:
+
+```
+    Prefix Verb   URI Pattern                          Controller#Action
+     users GET    /users(.:format)                     users#index
+           POST   /users(.:format)                     users#create
+  new_user GET    /users/new(.:format)                 users#new
+ edit_user GET    /users/:id/edit(.:format)            users#edit
+      user GET    /users/:id(.:format)                 users#show
+           PATCH  /users/:id(.:format)                 users#update
+           PUT    /users/:id(.:format)                 users#update
+           DELETE /users/:id(.:format)                 users#destroy
+```
+
+As can be seen, the common routes that you would need for the resources have already been mapped to their respective URI's. This is equivalent to manually declaring the 7 resourceful routes for the `user` resource. By convention, Rails will then look for a controller called `users_controller.rb` and invoke the name of the function mapped to the particular URI and Verb. More information about Rails routes and how they work can be found [here](https://guides.rubyonrails.org/routing.html#resource-routing-the-rails-default).
+
+#### Controller
+In the standard Rails convention, once inside the application directory, if you navigate to `app/controllers/`, you should be able to see all the controllers in your project. Rails will look for the controller with the `<name>_controller.rb`, where `<name>` would be replaced by the result under the `Controller` heading in the list of routes above, in this case, `users`. It will then look for the function name under the `Action` heading mapped to the URI function inside the Controller, and then call it. For example, if someone makes a GET request to /users, Rails will then respond by calling the users#index controller action (refer to the code below). 
+
+```ruby
+class UsersController < ApplicationController
+  def index
+    ..
+  end
+end
+```
+<center>
+`users_controller.rb` <i>with an </i>`index` <i>method</i> 
+</center>
```

Comment:
What this currently looks like:

<img width="393" alt="Screenshot 2019-04-17 at 12 04 11 PM" src="https://user-images.githubusercontent.com/9073504/56260264-27b0aa00-6109-11e9-95bc-e0407133c46d.png">

Inserting a line break after the `<center>` tag should solve this.

Reply:
I don't think you verified this with the preview. Adding the `<br>` doesn't work. What I meant was:

```html
<center>

`users_controller.rb` <i>with an </i>`index` <i>method</i> 

</center>